src/lib/storage/table.cpp

```diff
@@ -432,8 +436,28 @@ void Table::add_soft_key_constraint(const TableKeyConstraint& table_key_constrai
   _table_key_constraints.insert(table_key_constraint);
 }
 
-const TableKeyConstraints& Table::soft_key_constraints() const {
-  return _table_key_constraints;
+void Table::delete_key_constraint(const TableKeyConstraint& constraint) {
+  _table_key_constraints.erase(constraint);
+}
+
+bool Table::constraint_guaranteed_to_be_valid(const TableKeyConstraint& table_key_constraint) const {
+  if (!table_key_constraint.can_become_invalid()) {
+    return true;
+  }
+
+  const auto chunk_count = this->chunk_count();
```

Comment:
Can `this->` be removed?

Reply:
No, if I try to remove `this->` I get the following error `Variable 'chunk_count' declared with deduced type 'const auto' cannot appear in its own initializer`.